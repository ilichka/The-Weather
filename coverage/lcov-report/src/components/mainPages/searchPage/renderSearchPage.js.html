
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/mainPages/searchPage/renderSearchPage.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="..\..\..\..\index.html">All files</a> / <a href="index.html">src/components/mainPages/searchPage</a> renderSearchPage.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>12/12</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>12/12</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import addListenerToInput from "./addListenerToInput";
&nbsp;
const renderInput = () =&gt; {
  const input = document.createElement("input");
  input.classList.add("search-input");
  input.minLength = 3;
  input.placeholder = "Search";
  addListenerToInput(input);
  return input;
};
&nbsp;
export default () =&gt; {
  const searchRow = document.createElement("div");
  searchRow.classList.add("search-row");
  searchRow.innerHTML = `&lt;svg class="icon" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"&gt;&lt;path d="M7.127 22.562l-7.127 1.438 1.438-7.128 5.689 5.69zm1.414-1.414l11.228-11.225-5.69-5.692-11.227 11.227 5.689 5.69zm9.768-21.148l-2.816 2.817 5.691 5.691 2.816-2.819-5.691-5.689z"/&gt;&lt;/svg&gt;`;
  searchRow.prepend(renderInput());
  return searchRow;
};
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Oct 27 2021 14:48:35 GMT+0300 (Москва, стандартное время)
            </div>
        </div>
        <script src="..\..\..\..\prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="..\..\..\..\sorter.js"></script>
        <script src="..\..\..\..\block-navigation.js"></script>
    </body>
</html>
    