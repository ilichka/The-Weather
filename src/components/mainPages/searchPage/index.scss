.search-page {
  padding: 16px 16px 0 16px;

  .search-row {
    display: flex;
    align-items: center;

    .search-input {
      flex-grow: 1;
      height: 48px;
      margin: 0 22px;
      border: 1px solid transparent;
      font-size: 40px;
      border-radius: 10px;
      padding: 5px;

      &:invalid:focus,
      &:invalid {
        border: 1px solid red;
      }

      &:focus-visible {
        outline: none;
      }
    }
  }

  .cards-block {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    gap: 12px;
    margin-top: 20px;

    .card {
      cursor: pointer;
      background-color: #07b299;
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      min-height: 186px;
      flex: 1 1 calc(100% / 2 - 12px);
      box-shadow: 0px 3px 2px rgb(30 79 68 / 5%),
        0px 10px 10px rgb(30 79 68 / 8%);
      border-radius: 5px;
      overflow: hidden;

      .top {
        padding: 16px 16px 0 16px;
        display: flex;
        justify-content: space-between;

        .weather-info {
          display: flex;
          flex-direction: column;
          color: white;

          .temp {
            font-size: 36px;
            margin-bottom: 5px;
          }

          .city {
            font-size: 20px;
            margin-bottom: 8px;
          }

          .country {
            font-family: "Roboto", sans-serif;
            font-size: 0.875rem;
            line-height: 1rem;
            font-weight: 500;
          }
        }

        .weather-icon {
          width: 60px;
          height: 60px;
        }
      }

      .bottom {
        padding: 0 16px 16px 16px;
        display: flex;
        justify-content: space-between;
        flex-direction: row;

        .humidity-info,
        .wind-info {
          display: flex;
          align-items: center;
        }

        .wind-icon,
        .humidity-icon {
          width: 24px;
          height: 24px;
          margin-right: 8px;
        }

        .wind-speed,
        .humidity {
          color: white;
          font-size: 22px;
        }

        .humidity-info {
        }
      }
    }

    .no-match {
      flex-grow: 1;
      text-align: center;
      font-family: "Roboto", sans-serif;
      font-size: 1.2rem;
      line-height: normal;
    }

    & > :last-child {
      flex-grow: 0;
      flex-basis: calc(50% - 8px);
    }
  }
}
