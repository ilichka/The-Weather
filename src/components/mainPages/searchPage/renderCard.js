import * as waterDropIcon from "../../../assets/img/waterdrop.svg";
import * as windIcon from "../../../assets/img/wind.svg";
import {
  getLocalStorageField,
  setLocalStorageField,
} from "../../../utils/localStorageManager";
import { LOCAL_STORAGE_KEYS } from "../../../utils/consts";

const addCardListener = (node, obj, id) => {
  node.addEventListener("click", () => {
    const history = JSON.parse(
      getLocalStorageField(LOCAL_STORAGE_KEYS.HISTORY)
    );
    const newHistory = history ? { ...history, [id]: obj } : { [id]: obj };
    const lastClicked = { [id]: obj };
    setLocalStorageField(
      LOCAL_STORAGE_KEYS.HISTORY,
      JSON.stringify(newHistory)
    );
    setLocalStorageField(
      LOCAL_STORAGE_KEYS.LAST_CLICKED,
      JSON.stringify(lastClicked)
    );
    location.hash = "more";
  });
};

const getCardHTML = (obj) => {
  const { city, country, temp, humidity, wind, icon } = obj;
  return `
            <div class="top">
              <div class="weather-info">
              <span class="temp">${temp}Â°</span>
              <span class="city">${city}</span>
              <span class="country">${country}</span></div>
              <img class="weather-icon" src="http://openweathermap.org/img/wn/${icon}.png" alt="icon"/>
            </div>
            <div class="bottom">
              <div class="humidity-info">
                <img class="humidity-icon" src="${waterDropIcon.default}" alt="humidity"/>
                <span class="humidity">${humidity}%</span>
              </div>
              <div class="wind-info">
                <img class="wind-icon" src="${windIcon.default}" alt="wind"/>
                <span class="wind-speed">${wind}km/h</span>
              </div>
            </div>
          `;
};

export default (obj, id) => {
  const cardWrapper = document.createElement("div");
  cardWrapper.classList.add("card");
  cardWrapper.innerHTML = getCardHTML(obj);
  addCardListener(cardWrapper, obj, id);
  return cardWrapper;
};
