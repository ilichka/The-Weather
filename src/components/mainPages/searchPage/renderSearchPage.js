const renderSearchPage = () => {
  return `
        <div class="search-row">
          <svg class="icon" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M9.145 18.29c-5.042 0-9.145-4.102-9.145-9.145s4.103-9.145 9.145-9.145 9.145 4.103 9.145 9.145-4.102 9.145-9.145 9.145zm0-15.167c-3.321 0-6.022 2.702-6.022 6.022s2.702 6.022 6.022 6.022 6.023-2.702 6.023-6.022-2.702-6.022-6.023-6.022zm9.263 12.443c-.817 1.176-1.852 2.188-3.046 2.981l5.452 5.453 3.014-3.013-5.42-5.421z"/></svg>
          <input placeholder="Search" class="search-input" minlength="3"/>
          <svg class="icon" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M7.127 22.562l-7.127 1.438 1.438-7.128 5.689 5.69zm1.414-1.414l11.228-11.225-5.69-5.692-11.227 11.227 5.689 5.69zm9.768-21.148l-2.816 2.817 5.691 5.691 2.816-2.819-5.691-5.689z"/></svg>
        </div>
        <div class="cards-block"></div>`;
};

export default renderSearchPage;
