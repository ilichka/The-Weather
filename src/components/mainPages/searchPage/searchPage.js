import "./index.scss";
import * as waterDropIcon from "../../../assets/img/waterdrop.svg";
import * as windIcon from "../../../assets/img/wind.svg";
import * as loupeIcon from "../../../assets/img/loupe.svg";
import * as pencilIcon from "../../../assets/img/pencil.svg";
import errorHandler from "../../../utils/utils";

let timeout;

const createCard = (obj, id) => {
  const { city, country, temp, humidity, wind, icon } = obj;
  const card = `
            <div class="top">
              <div class="weather-info">
              <span class="temp">${temp}Â°</span>
              <span class="city">${city}</span>
              <span class="country">${country}</span></div>
              <img class="weather-icon" src="http://openweathermap.org/img/wn/${icon}.png" alt="icon"/>
            </div>
            <div class="bottom">
              <div class="humidity-info">
                <img class="humidity-icon" src="${waterDropIcon.default}" alt="humidity"/>
                <span class="humidity">${humidity}%</span>
              </div>
              <div class="wind-info">
                <img class="wind-icon" src="${windIcon.default}" alt="wind"/>
                <span class="wind-speed">${wind}km/h</span>
              </div>
            </div>
          `;
  const cardd = document.createElement("div");
  cardd.classList.add("card");
  cardd.innerHTML = card;
  cardd.addEventListener("click", () => {
    const history = JSON.parse(localStorage.getItem("history"));
    const newHistory = history ? { ...history, [id]: obj } : { [id]: obj };
    const lastClicked = { [id]: obj };
    localStorage.setItem("history", JSON.stringify(newHistory));
    localStorage.setItem("lastClicked", JSON.stringify(lastClicked));
    location.hash = "more";
  });
  document.querySelector(".cards-block").appendChild(cardd);
};

const getCityList = (cityName) => {
  const key = "d0b9cf5011a1d4ea6ac31f2492fda53d";
  fetch(
    `https://api.openweathermap.org/data/2.5/find?q=${cityName}&type=like&units=metric&sort=population&cnt=3&appid=${key}`
  )
    .then((resp) => resp.json()) // Convert data to json
    .then((data) => {
      if (+data.cod === 200) {
        document.querySelector(".cards-block").innerHTML = ``;
        if (data.list.length) {
          data.list.forEach((elem) => {
            createCard(
              {
                city: elem.name,
                country: elem.sys.country,
                temp: Math.round(elem.main.temp),
                humidity: elem.main.humidity,
                wind: elem.wind.speed,
                icon: elem.weather[0].icon,
                status: elem.weather[0].description,
              },
              String(elem.id)
            );
          });
        } else {
          document.querySelector(
            ".cards-block"
          ).innerHTML = `<span class="no-match">There is no such city:c</span>`;
        }
      } else {
        errorHandler(data.message);
      }
    })
    .catch(() => {
      errorHandler(`Something went wrong`);
    });
};

const addListenerToInput = () => {
  document.querySelector(".search-input").addEventListener("input", (e) => {
    clearTimeout(timeout);
    timeout = window.setTimeout(() => {
      getCityList(e.target.value);
      clearTimeout(timeout);
    }, 2000);
  });
  document.querySelector(".search-input").addEventListener("keydown", (e) => {
    if (e.code === "Enter") {
      clearTimeout(timeout);
      getCityList(e.target.value);
      document.querySelector(".search-input").blur();
    }
  });
};

const searchPage = () => {
  const page = document.createElement("div");
  page.classList.add("search-page");
  page.innerHTML = `
        <div class="search-row">
          <img class="icon" src="${loupeIcon.default}" alt="loupe"/>
          <input placeholder="Search" class="search-input" minlength="3"/>
          <img class="icon" src="${pencilIcon.default}" alt="pencil"/>
        </div>
        <div class="cards-block"></div>`;
  document.querySelector(".app").appendChild(page);
  const history = JSON.parse(localStorage.getItem("history")) || {};
  if (Object.keys(history).length !== 0) {
    for (const prop in history) {
      createCard(history[prop], prop);
    }
  } else {
    document.querySelector(
      ".cards-block"
    ).innerHTML = `<span class="no-match">Your history is clean:c</span>`;
  }
  addListenerToInput();
};

export default searchPage;
