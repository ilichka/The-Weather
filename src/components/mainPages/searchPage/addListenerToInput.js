import { getCityListByName } from "../../../requester";
import {
  errorHandler,
  clearCardsBlock,
  setNoMatchHTML,
} from "../../../utils/utils";
import renderCard from "./renderCard";

let timeout;

const callBackFunction = (data) => {
  console.log(data);
  if (+data.cod === 200) {
    clearCardsBlock();
    if (data.list.length) {
      data.list.forEach((elem) => {
        renderCard(
          {
            city: elem.name,
            country: elem.sys.country,
            temp: Math.round(elem.main.temp),
            humidity: elem.main.humidity,
            wind: elem.wind.speed,
            icon: elem.weather[0].icon,
            status: elem.weather[0].description,
          },
          String(elem.id)
        );
      });
    } else {
      setNoMatchHTML(`There is no such city:c`);
    }
  } else {
    errorHandler(data.message);
  }
};

const addListenerToInput = () => {
  document.querySelector(".search-input").addEventListener("input", (e) => {
    clearTimeout(timeout);
    timeout = window.setTimeout(() => {
      getCityListByName(e.target.value, callBackFunction);
      clearTimeout(timeout);
    }, 2000);
  });
  document.querySelector(".search-input").addEventListener("keydown", (e) => {
    if (e.code === "Enter") {
      clearTimeout(timeout);
      getCityListByName(e.target.value, callBackFunction);
      document.querySelector(".search-input").blur();
    }
  });
};

export default addListenerToInput;
