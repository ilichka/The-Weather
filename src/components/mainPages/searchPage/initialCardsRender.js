import renderCard from "./renderCard";
import { setNoMatchHTML, clearCardsBlock } from "../../../utils/utils";

const initialCardsRender = () => {
  clearCardsBlock();
  const history = JSON.parse(localStorage.getItem("history")) || {};
  if (Object.keys(history).length !== 0) {
    for (const prop in history) {
      renderCard(history[prop], prop);
    }
  } else {
    setNoMatchHTML(`Your history is clean:c`);
  }
};

export default initialCardsRender;
