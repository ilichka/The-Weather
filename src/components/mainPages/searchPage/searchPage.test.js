import searchPage from "./searchPage";

describe("Search page snapshot", () => {
  it("Renders correctly template", () => {
    const template = searchPage();

    expect(template).toMatchSnapshot();
  });
});
