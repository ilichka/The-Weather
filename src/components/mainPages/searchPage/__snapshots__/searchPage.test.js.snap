// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page snapshot Renders correctly template 1`] = `
<div
  class="search-page"
>
  <div
    class="search-row"
  >
    <input
      class="search-input"
      minlength="3"
      placeholder="Search"
    />
    <svg
      class="icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.127 22.562l-7.127 1.438 1.438-7.128 5.689 5.69zm1.414-1.414l11.228-11.225-5.69-5.692-11.227 11.227 5.689 5.69zm9.768-21.148l-2.816 2.817 5.691 5.691 2.816-2.819-5.691-5.689z"
      />
    </svg>
  </div>
  <div
    class="cards-block"
    id="cards-block"
  >
    <span
      class="no-match"
    >
      Your history is clean:c
    </span>
  </div>
</div>
`;
